### PR TITLE
Modify current dropdown to more flexible

### PR DESCRIPTION
**Pull request by JustinBeBoy**
This PR custom the dropdown with feature:
- Add separator line
- Support add superview
- Add dismissview
- Add sublabel
- Recalculate coordinates, with, height

Screenshot:
Screenshot 2021-03-03 at 17 58 13

### CODE REVIEW

File: Decred Wallet/view_util/DropMenuButton.swift [hunk 1]
```diff
@@ -104,22 +131,35 @@ class DropMenuButton: UIButton, UITableViewDelegate, UITableViewDataSource
...
         self.setTitle(self.selectedItem!, for: .normal)
         self.setTitle(self.selectedItem!, for: .selected)
         self.setTitle(self.selectedItem!, for: .highlighted)
 
         table.reloadData()
     }
     
+    func caculateWidthCell() {
+        self.items.forEach { (item) in
+            let widthText = item.text.width(withConstrainedHeight: 20, font: UIFont(name: "SourceSansPro-Regular", size: 17) ?? UIFont.systemFont(ofSize: 17))
```
Comment (macsleven) [inline Decred Wallet/view_util/DropMenuButton.swift:144]: value of type string has no member width

File: Decred Wallet/view_util/DropMenuButton.swift [hunk 2]
```diff
@@ -104,22 +131,35 @@ class DropMenuButton: UIButton, UITableViewDelegate, UITableViewDataSource
...
+            let widthtextLabel = item.textLabel.width(withConstrainedHeight: 20, font: UIFont(name: "SourceSansPro-Bold", size: 13) ?? UIFont.systemFont(ofSize: 13)) + 10
```
Comment (macsleven) [inline Decred Wallet/view_util/DropMenuButton.swift:145]: value of type string has no member width

**Review by macsleven — CHANGES_REQUESTED** [review]

File: Decred Wallet/view_util/DropMenuButton.swift (hunk 1 above)
Comment (JustinBeBoy) [reply Decred Wallet/view_util/DropMenuButton.swift:144]: Of course it have

File: Decred Wallet/view_util/DropMenuButton.swift (hunk 2 above)
Comment (JustinBeBoy) [reply Decred Wallet/view_util/DropMenuButton.swift:145]: Of course it have

File: Decred Wallet/view_util/DropMenuButton.swift (hunk 1 above)
Comment (C-ollins) [reply Decred Wallet/view_util/DropMenuButton.swift:144]: I get this error when I try to compile `Value of type 'String' has no member 'width'`. @JustinBeBoy confirm that you have the current implementation.

**Review by macsleven — CHANGES_REQUESTED** [review]
Implement this in wallet page

**Review by macsleven — CHANGES_REQUESTED** [review]
- The dropdown menu should dismiss when you tap outside it.
- Implement same dropdown menu for watch only wallet(without the privacy menu in the list.)
- Use the appropriate font for the list.
- There should be the new label beside the privacy label when privacy is not setup.

Screenshot 2021-05-18 at 23 41 48

**Review by macsleven — CHANGES_REQUESTED** [review]
The new label should not display wen privacy is already setup.
Dismiss the dropdown when the view disappears.

**Review by C-ollins — CHANGES_REQUESTED** [review]
Rebase your code with master so that you can have this commit https://github.com/planetdecred/dcrios/pull/762
Simiar to #762, the `new` text should not be shown in the list once the user views privacy page or privacy setup page.

**Review by macsleven — CHANGES_REQUESTED** [review]
The selection color is off make it lighter
Screenshot 2021-05-30 at 12 46 59

The navigation from the wallet page is wrong, it should only display the input dialog.
Screenshot 2021-05-30 at 12 49 46
Screenshot 2021-05-30 at 12 50 48
Screenshot 2021-05-30 at 12 51 50

**Comment by JustinBeBoy** [comment]
> The selection color is off make it lighter
> Screenshot 2021-05-30 at 12 46 59

it's exactly Lightgrey

**Comment by JustinBeBoy** [comment]
> 
> The navigation from the wallet page is wrong, it should only display the input dialog.
> Screenshot 2021-05-30 at 12 49 46
> Screenshot 2021-05-30 at 12 50 48
> Screenshot 2021-05-30 at 12 51 50
I don't get it, please check it again

**Comment by dreacot** [comment]
> > The selection color is off make it lighter
> > Screenshot 2021-05-30 at 12 46 59
> 
> it's exactly Lightgrey

make the grey a lighter shade, the current gray is too dark, maybe a similar gray to a disabled button

**Comment by dreacot** [comment]
Implement the dropdown for this menu also

**Comment by dreacot** [comment]
The dropdown covers the balance when open, it should be like in the screenshot below

**Review by macsleven — CHANGES_REQUESTED** [review]
The dropdown menu for watch only wallet is smaller than for normal wallet 
Screenshot 2021-05-31 at 21 51 03 Screenshot 2021-05-31 at 21 51 22

**Review by macsleven — APPROVED** [review]

**Pull request merged**